examples/Accelerators/DarcyAccelerated/darcy_accelerated.jl

```diff
@@ -0,0 +1,315 @@
+    pts_per_dim,
+)
+
+# We define a wrapper around the GRF, and as the permeability field must be positive we introduce a domain constraint into the function distribution. Henceforth, the GRF is interfaced in the same manner as any other parameter distribution with regards to interface.
+pkg = GRFJL()
+distribution = GaussianRandomFieldInterface(grf, pkg) # our wrapper from EKP
+domain_constraint = bounded_below(0) # make κ positive
+pd = ParameterDistribution(Dict("distribution" => distribution, "name" => "kappa", "constraint" => domain_constraint)) # the fully constrained parameter distribution
+
+# Now we have a function distribution, we sample a reasonably high-probability value from this distribution as a true value (here all degrees of freedom set with `u_{\mathrm{true}} = -0.5`). We use the EKP transform function to build the corresponding instance of the ``\kappa_{\mathrm{true}}``.
+u_true = -1.5 * ones(dofs, 1) # the truth parameter
+κ_true = transform_unconstrained_to_constrained(pd, u_true) # builds and constrains the function.  
+κ_true = reshape(κ_true, N, N)
+
+# Now we generate the data sample for the truth in a perfect model setting by evaluating the the model here, and observing it by subsampling in each dimension every `obs_ΔN` points, and add some observational noise
+darcy = Setup_Param(pts_per_dim, obs_ΔN, κ_true)
+h_2d = solve_Darcy_2D(darcy, κ_true)
+y_noiseless = compute_obs(darcy, h_2d)
+obs_noise_cov = 0.05^2 * I(length(y_noiseless)) * (maximum(y_noiseless) - minimum(y_noiseless))
+truth_sample = vec(y_noiseless + rand(rng, MvNormal(zeros(length(y_noiseless)), obs_noise_cov)))
+
+# Now we set up the Bayesian inversion algorithm. The prior we have already defined to construct our truth
+prior = pd
+
+# We define some algorithm parameters, here we take ensemble members larger than the dimension of the parameter space
+N_ens = dofs + 2    # number of ensemble members
+N_iter = 20         # number of EKI iterations
+N_trials = 5       # number of trials 
+
+"""
+Define function run_darcy() with parameter ACCELERATED to solve the Darcy problem using either
+the default EKI algorithm (ACCELERATED=TRUE) or the EKI algorithm with Nesterov acceleration (ACCELERATED=FALSE).
+
+Returns a vector of errors evaluated after each EKI iteration.
+"""
+function run_darcy(ACCELERATED::Bool)
+    errs = zeros(N_trials, N_iter)
+
+    # can make separate RNG for each thread, run with multithreading
+
+    for trial in 1:N_trials
+        # We sample the initial ensemble from the prior, and create the EKP object as an EKI algorithm using the `Inversion()` keyword
+        initial_params = construct_initial_ensemble(rng, prior, N_ens) 
+        if ACCELERATED
+            ekiobj = EKP.EnsembleKalmanProcess(
+                initial_params,
+                truth_sample,
+                obs_noise_cov,
+                Inversion(),
+                accelerator = NesterovAccelerator(),
+                scheduler = DefaultScheduler(0.5),
```

Comment:
Can we have `scheduler = deepcopy(scheduler)`

with scheduler defined outside these loops (and could you do this for all examples please!)